docs/development/go/go-context.md

```diff
@@ -0,0 +1,468 @@
+f2(): context deadline exceeded
+f3(): 2019-05-29 08:39:56.224505 +0300 EEST m=+12.009254674
+{{< /output >}}
+
+The long lines of the output are the return values of the `time.After()` function calls.
+They denote normal operation of the program. The point here is that the operation of the
+program is cancelled when there are delays in its execution.
+
+If you use a bigger delay (*10 seconds*), which is executed as a call to `time.Sleep()`, you
+will get the following kind of output:
+
+    go run simple.go 10
+{{< output >}}
+Delay: 10
+f1(): context canceled
+f2(): context canceled
+f3(): context canceled
+{{< /output >}}
+
+The calls to `time.Sleep()` simulate a program that is slow or an operation that takes too
+much time to finish. Production code does not usually have such `time.Sleep()` function calls.
+
+## Using Context for HTTP
+
+In this section of the guide you will learn how to timeout HTTP connections on the client side.
+
+The presented utility, which is called `http.go`, requires two command line arguments,
+which are the URL to connect to and the allowed delay value in seconds.
+
+### Explaining the Go Code of the Example
+
+The Go code of the `http.go` utility is the following:
+
+{{< file "./http.go" go >}}
+package main
+
+import (
+        "context"
+        "fmt"
+        "io/ioutil"
+        "net/http"
+        "os"
+        "strconv"
+        "sync"
+        "time"
+)
+
+var (
+        myUrl string
+        delay int = 5
+        w     sync.WaitGroup
+)
+
+type myData struct {
+        r   *http.Response
+        err error
+}
+
+func connect(c context.Context) error {
+        defer w.Done()
+        data := make(chan myData, 1)
+        tr := &http.Transport{}
+        httpClient := &http.Client{Transport: tr}
+        req, _ := http.NewRequest("GET", myUrl, nil)
+
+        go func() {
+                response, err := httpClient.Do(req)
+                if err != nil {
+                        fmt.Println(err)
+                        data <- myData{nil, err}
+                        return
+                } else {
+                        pack := myData{response, err}
+                        data <- pack
+                }
+        }()
+
+        select {
+        case <-c.Done():
+                tr.CancelRequest(req)
+                <-data
+                fmt.Println("The request was cancelled!")
+                return c.Err()
+        case ok := <-data:
+                err := ok.err
+                resp := ok.r
+                if err != nil {
+                        fmt.Println("Error select:", err)
+                        return err
+                }
+                defer resp.Body.Close()
+
+                realHTTPData, err := ioutil.ReadAll(resp.Body)
+                if err != nil {
+                        fmt.Println("Error select:", err)
+                        return err
+                }
+                fmt.Printf("Server Response: %s\n", realHTTPData)
```

Comment:
Why not use the `log` package here? It's sort of a minor point, but helps readers get in the habit of privileging `log` over `fmt` for debugging.

Reply:
That is a good point - I want the reader to be able to see the output. I will include what you said as you are right about it.